### ISSUE TITLE
Class computed fields incorrect transpilation

### ISSUE THREAD

**Issue by vemoo** [evanw/esbuild#1507]
```js
class Test {
    ["foo"] = 2;
}
```
get transpiled to
```js
var _a;
var Test = class {
  [_a] = 2;
};
_a = "foo";
```
which gets evaluated as
```js
var Test = class {
  [undefined] = 2;
};
```

**Comment by hyrious** [comment]
In addition, this behavior occurs in `--loader=ts` and `tsx`.

**Comment by vemoo** [comment]
Ok, this is a bit weird, it happens for `.ts` files when there's a `tsconfig.json` with `"target": "esnext"` but not when passing `--target=esnext`

**Comment by evanw** [comment]
This is likely a regression due to the recent breaking change that TypeScript made (and that esbuild picked up) where `useDefineForClassFields` defaults to true when `"target": "esnext"` is present. This means class fields are no longer initialized inside of the constructor, so it's no longer correct to defer the initialization of the field.

**Comment by hyrious** [comment]
This behavior is published in 4.3, as a reference: microsoft/TypeScript#42663. TypeScript tries to work in the same way as vanilla js does.

**Comment by hyrious** [comment]
@evanw Despite of this behavior, lifting up computed fields also causes this issue.
But it can be fixed by `--minify-syntax`. Do you think it can be avoided?

[Live Example](https://hyrious.me/esbuild-repl/?version=0.12.18&shareable=JTdCJTIyY29kZSUyMiUzQSUyMmNsYXNzJTIwVGVzdCUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlNUIlNUMlMjJmb28lNUMlMjIlNUQlMjAlM0QlMjAyJTNCJTVDbiU3RCUyMiUyQyUyMmNvbmZpZyUyMiUzQSU3QiUyMmxvYWRlciUyMiUzQSUyMnRzJTIyJTJDJTIydHNjb25maWdSYXclMjIlM0ElMjIlN0IlNUMlMjJjb21waWxlck9wdGlvbnMlNUMlMjIlM0ElN0IlNUMlMjJ1c2VEZWZpbmVGb3JDbGFzc0ZpZWxkcyU1QyUyMiUzQXRydWUlN0QlN0QlMjIlMkMlMjJtaW5pZnlTeW50YXglMjIlM0F0cnVlJTdEJTdE).
